e2e/staticpvc.go

```diff
@@ -159,3 +159,116 @@ func validateRBDStaticPV(f *framework.Framework, appPath string, isBlock bool) e
 	execCommandInPod(f, cmd, rookNamespace, &listOpt)
 	return nil
 }
+
+func validateCephFsStaticPV(f *framework.Framework, appPath, scPath string) error {
+	opt := make(map[string]string)
+	var (
+		cephFsVolName = "staticCephFsVol"
+		groupName     = "e2e"
+		fsName        = "myfs"
+		pvName        = "pv-name"
+		pvcName       = "pvc-name"
+		// minikube creates default class in cluster, we need to set dummy
+		// storageclass on PV and PVC to avoid storageclass name missmatch
+		sc = "storage-class"
+	)
+	const secretName = "cephfs-static-pv-sc" // #nosec
+
+	c := f.ClientSet
+
+	listOpt := metav1.ListOptions{
+		LabelSelector: "app=rook-ceph-tools",
+	}
+
+	fsID, e := execCommandInPod(f, "ceph fsid", rookNamespace, &listOpt)
+	if e != "" {
+		return fmt.Errorf("failed to get fsid from ceph cluster %s", e)
+	}
+	// remove new line present in fsID
+	fsID = strings.Trim(fsID, "\n")
+
+	// 4G in bytes
+	size := "4294967296"
+
+	// create subvolumegroup, command will work even if group is already present.
+	cmd := fmt.Sprintf("ceph fs subvolumegroup create %s %s", fsName, groupName)
+
+	_, e = execCommandInPod(f, cmd, rookNamespace, &listOpt)
+	if e != "" {
+		return fmt.Errorf("failed to create subvolumegroup %s", e)
+	}
+
+	// create subvolume
+	cmd = fmt.Sprintf("ceph fs subvolume create %s %s %s --size %s", fsName, cephFsVolName, groupName, size)
+	_, e = execCommandInPod(f, cmd, rookNamespace, &listOpt)
+	if e != "" {
+		return fmt.Errorf("failed to create subvolume %s", e)
+	}
+
+	opt["clusterID"] = fsID
+	opt["fsName"] = fsName
+	opt["staticVolume"] = "true"
+	opt["rootPath"] = "/volumes/" + groupName + "/" + cephFsVolName
+
+	// create secret
+	const userID = "admin"
+	secret := getSecret(scPath)
+	adminKey, e := execCommandInPod(f, "ceph auth get-key client.admin", rookNamespace, &listOpt)
+	if e != "" {
+		return fmt.Errorf("failed to get adminKey %s", e)
+	}
+	secret.StringData["userID"] = userID
+	secret.StringData["userKey"] = adminKey
+	secret.Name = secretName
+	_, err := c.CoreV1().Secrets("default").Create(context.TODO(), &secret, metav1.CreateOptions{})
+	if err != nil {
+		return fmt.Errorf("failed to create secret, error %v", err)
+	}
+
+	pv := getStaticPV(pvName, pvName, "4Gi", secretName, "default", sc, "cephfs.csi.ceph.com", false, opt)
+	_, err = c.CoreV1().PersistentVolumes().Create(context.TODO(), pv, metav1.CreateOptions{})
+	if err != nil {
+		return fmt.Errorf("failed to create PV, error %v", err)
+	}
+
+	pvc := getStaticPVC(pvcName, pvName, size, "default", sc, false)
+
+	_, err = c.CoreV1().PersistentVolumeClaims(pvc.Namespace).Create(context.TODO(), pvc, metav1.CreateOptions{})
+	if err != nil {
+		return fmt.Errorf("failed to create PVC, error %v", err)
+	}
+	// bind pvc to app
+	app, err := loadApp(appPath)
+	if err != nil {
+		return fmt.Errorf("failed to load app, error %v", err)
+	}
+
+	app.Namespace = "default"
```

Comment:
create an app in cephCSINamespace namespace. same for PVC also. the resources we are creating should be in `cephCSINamespace` namespace